lib/python/qmk/cli/format/json.py

```diff
@@ -92,8 +92,8 @@ def format_json(cli):
     output = json.dumps(json_data, cls=json_encoder, sort_keys=True)
 
     if cli.args.inplace:
-        with open(cli.args.json_file, 'w+', encoding='utf-8') as outfile:
-            outfile.write(output)
+        with open(cli.args.json_file, 'w+', encoding='utf-8', newline='\n') as outfile:
+            outfile.write(output + '\n')
```

Comment:
If you run it multiple times does it add a new one each time?

Reply:
Nope, because the command has to deserialise it with `json_load()` in order to re-serialise it.
It's possible to just do this directly in `QMKJSONEncoder` since it knows when it's encoding the root object:
```python
return "{\n" + ",\n".join(output) + "\n" + self.indent_str + "}" + ("\n" if len(path) == 0 else "")
```
but when using `-p` instead of `-i`, the `print()` gives it an extra newline...six of one and all that.